lib/icalendar/parser.rb

```diff
@@ -23,6 +23,7 @@ def parse
       read_in_data
       components = []
       while (fields = next_fields)
+        component = component_class.new
```

Comment:
Hmm, that'd require switching around `component.ical_name` as well. Maybe keep this version.